peerdid/peer_did_utils.py

```diff
@@ -0,0 +1,266 @@
+import base64
+import hashlib
+import json
+import re
+from typing import Union, List
+
+import base58
+import varint
+
+from peerdid.types import JSON, PublicKeyAgreement, PublicKeyAuthentication, PublicKeyTypeAgreement, \
+    PublicKeyTypeAuthentication, PEER_DID, EncodingType
+
+
+def _encode_service(service: JSON) -> str:
+    """
+    Generates encoded service according to the second algorithm
+    (https://identity.foundation/peer-did-method-spec/index.html#generation-method)
+    For this type of algorithm did_doc can be obtained from peer_did
+
+    :param service: JSON string conforming to the DID specification (https://www.w3.org/TR/did-core/#services)
+
+    :return: encoded service
+    """
+    service_to_encode = re.sub(r"[\n\t\s]*", "", service) \
+        .replace('\'', '\"') \
```

Comment:
A bit of an edge case, but could this have implications on something like
```
{
  "serviceEndpoint": "https://myurl.com?q=some 'thing'"
}
```

Perhaps they should be encoded beforehand, but I _think_ https://www.rfc-editor.org/rfc/rfc3986#section-2 -> "2.2.  Reserved Characters" allows it

Reply:
Thanks, i will add encoding and decoding of url special characters